File: plugins/plugin-demo/src/main/java/com/voc/fr/plugin/api/ITableMeta.java
//package com.voc.fr.plugin.api;
//
//import com.voc.fr.plugin.data.html.IRowMeta;
//
//import java.util.Collection;
//
///**
// * 解析生成最终二维表信息
// *
// * @author Wu Yujie
// * @email coffee377@dingtalk.com
// * @time 2019/09/04 22:00
// */
//public interface ITableMeta {
//
//    /**
//     * 表格名称
//     *
//     * @return String
//     */
//    String getName();
//
//    /**
//     * 数据行信息
//     *
//     * @return Collection<IRowMeta>
//     */
//    Collection<IRowMeta> getRowMeta();
//
//    /**
//     * 行数（合并单元格拆分后）
//     *
//     * @return int
//     */
//    int getRowCount();
//
//    /**
//     * 列数（合并单元格拆分后）
//     *
//     * @return int
//     */
//    int getColumnCount();
//
//    /**
//     * 解析后原始数据（含字段名）
//     *
//     * @return String[][]
//     */
//    Object[][] rawData();
//
//    /**
//     * 字段名
//     *
//     * @param fieldRow 字段所在行
//     * @return String[]
//     */
//    String[] getFieldName(int fieldRow);
//
//    /**
//     * 实际展示数据
//     *
//     * @param fieldRow 字段所在行
//     * @return String[][]
//     */
//    Object[][] getData(int fieldRow);
//
//}

File: plugins/plugin-demo/src/main/kotlin/com/voc/fr/plugin/data/html/ITableMeta.kt
package com.voc.fr.plugin.data.html

/**
 * 二维表信息
 *
 * @author Wu Yujie
 * @email coffee377@dingtalk.com
 * @time 2019/09/04 22:00
 */
interface ITableMeta {

    /**
     * 表名称
     *
     * @return String
     */
    val name: String?

    /**
     * 数据行信息
     *
     * @return Collection<IRowMeta>
     */
    val rowMetas: Collection<IRowMeta>

    /**
     * 行数（合并单元格拆分后）
     *
     * @return int
     */
    val rowCount: Int

    /**
     * 列数（合并单元格拆分后）
     *
     * @return int
     */
    val columnCount: Int

    /**
     * 数据存储集合
     */
    val dataStorageSet:Array<Array<Any?>>

    /**
     * 字段名
     *
     * @param fieldRow 字段所在行
     * @return String[]
     */
    fun getFieldName(fieldRow: Int): Array<String>

    /**
     * 实际展示数据
     *
     * @param fieldRow 字段所在行
     * @return String[][]
     */
    fun getData(fieldRow: Int): Array<Array<Any?>>

}